include/openthread/border_router.h

```diff
@@ -0,0 +1,172 @@
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the OpenThread Border Router API.
+ */
+
+#ifndef OPENTHREAD_BORDER_ROUTER_H_
+#define OPENTHREAD_BORDER_ROUTER_H_
+
+#include <openthread/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-thread-general
+ *
+ * @{
+ *
+ */
+
+/**
+ * This method provides a full or stable copy of the local Thread Network Data.
+ *
+ * @param[in]     aInstance    A pointer to an OpenThread instance.
+ * @param[in]     aStable      TRUE when copying the stable version, FALSE when copying the full version.
+ * @param[out]    aData        A pointer to the data buffer.
+ * @param[inout]  aDataLength  On entry, size of the data buffer pointed to by @p aData.
+ *                             On exit, number of copied bytes.
+ */
+OTAPI otError OTCALL otBorderRouterGetLocalNetData(otInstance *aInstance, bool aStable, uint8_t *aData,
+                                                   uint8_t *aDataLength);
+
+/**
+ * Add a border router configuration to the local network data.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ * @param[in]  aConfig   A pointer to the border router configuration.
+ *
+ * @retval OT_ERROR_NONE          Successfully added the configuration to the local network data.
+ * @retval OT_ERROR_INVALID_ARGS  One or more configuration parameters were invalid.
+ * @retval OT_ERROR_NO_BUFS       Not enough room is available to add the configuration to the local network data.
+ *
+ * @sa otRemoveBorderRouter
+ * @sa otSendServerData
+ */
+OTAPI otError OTCALL otBorderRouterAddOnMeshPrefix(otInstance *aInstance, const otBorderRouterConfig *aConfig);
+
+/**
+ * Remove a border router configuration from the local network data.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ * @param[in]  aPrefix   A pointer to an IPv6 prefix.
+ *
+ * @retval OT_ERROR_NONE  Successfully removed the configuration from the local network data.
+ *
+ * @sa otAddBorderRouter
+ * @sa otSendServerData
+ */
+OTAPI otError OTCALL otBorderRouterRemoveOnMeshPrefix(otInstance *aInstance, const otIp6Prefix *aPrefix);
+
+/**
+ * This function gets the next On Mesh Prefix in the local Network Data.
+ *
+ * @param[in]     aInstance  A pointer to an OpenThread instance.
+ * @param[inout]  aIterator  A pointer to the Network Data iterator context. To get the first on-mesh entry
+                             it should be set to OT_NETWORK_DATA_ITERATOR_INIT.
+ * @param[out]    aConfig    A pointer to where the On Mesh Prefix information will be placed.
+ *
+ * @retval OT_ERROR_NONE       Successfully found the next On Mesh prefix.
+ * @retval OT_ERROR_NOT_FOUND  No subsequent On Mesh prefix exists in the Thread Network Data.
+ *
+ */
+OTAPI otError OTCALL otBorderRouterGetNextLocalOnMeshPrefix(otInstance *aInstance, otNetworkDataIterator *aIterator,
```

Comment:
I think the `otBorderRouter*` module should be limited to modifying local network data and registering that local network data with the leader.  If that is the case, then we can remove `Local` from the `otBorderRouter*` calls.  In this case, we can simplify to `otBorderRouterGetNextOnMeshPrefix(...)`.  Thoughts?